scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/GitRepositoryService.java

```diff
@@ -0,0 +1,110 @@
+import java.util.ArrayList;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.scriptoocloud.core.data.GitRepositoryRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.model.GitRepository;
+import edu.asu.diging.scriptoocloud.core.model.impl.GitRepositoryImpl;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.GitRepositoryManager;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+import edu.asu.diging.scriptoocloud.core.service.UrlFormatterUtility;
+
+/*
+ * Clones remote git repositories and facilitates transactional events related to git repositories in file system
+ * 
+ * @author Jason Ormsby
+ */
+
+@Service
+@Transactional
+@PropertySource("classpath:config.properties")
+public class GitRepositoryService implements GitRepositoryManager{
+
+    @Autowired
+    private DeleteFilesService deleteFilesService;
+    
+    @Autowired
+    private UrlFormatterUtility urlFormatter;
+    
+    @Autowired 
+    private GitRepositoryRepository gitRepositoryJpa;
+    
+    @Autowired
+    private JgitService jGitService;
+
+    @Value("${git.repositories.path}")
+    public String path;
+
+   /*
+    * Handles requests to clone remote git repositories and stores an entity, or updates an entity, 
+    * with information related to the request and repository
+    * 
+    * @param   gitUrl       nonmalformed url of remote git repository 
+    * @param   requester    username in current session that made the request
+    */
+    @Override
+    public void cloneRepository(String gitUrl, String requester) throws InvalidGitUrlException, MalformedURLException{
+        String folderName = urlFormatter.urlToFolderName(gitUrl);
+
+        ZonedDateTime creationDate = ZonedDateTime.now();       
+        
+        GitRepositoryImpl repositoryEntity = gitRepositoryJpa.findByUrl(gitUrl);  
+        
+        if(repositoryEntity != null){
+            repositoryEntity.setUrl(gitUrl);
+            repositoryEntity.setRequester(requester);
+            repositoryEntity.setCreationDate(creationDate);
+            repositoryEntity.setFolderName(folderName);
+            File file = new File(path + repositoryEntity.getFolderName());
+            deleteFilesService.deleteDirectoryContents(file);
+        }                          
+        else{
+            repositoryEntity = new GitRepositoryImpl();
+            repositoryEntity.setUrl(gitUrl);
+            repositoryEntity.setRequester(requester);
+            repositoryEntity.setCreationDate(creationDate);
+            repositoryEntity.setFolderName(folderName);
```

Comment:
there are several lines of duplicate code here, should be DRY (don't repeat yourself)